src/plugins/simulator/visualizations/qt-opengl/qtopengl_camera.h

```diff
@@ -37,85 +37,54 @@ namespace argos {
       struct SPlacement {
          /** The position of the camera in the global reference frame */
          CVector3 Position;
-         /** The local Y axis of the camera in the global reference frame */
-         CVector3 Left;
+         /** What we are looking at in the global reference frame */
+         CVector3 Target;
          /** The local Z axis of the camera in the global reference frame */
          CVector3 Up;
-         /** The local X axis of the camera in the global reference frame */
-         CVector3 Forward;
-         /** The direction of sight of the camera in the global reference frame */
-         CVector3 Target;
          /** The focal length of the lens (if this was a real camera) */
          Real LensFocalLength;
          /** The focal length of the camera */
          CDegrees YFieldOfView;
-         /** Motion sensitivity */
-         Real MotionSensitivity;
-         /** Rotation sensitivity */
-         Real RotationSensitivity;
-
-         SPlacement() :
-            Position(-2.0f, 0.0f, 2.0f),
-            Left(CVector3::Y),
-            Up(CVector3(1.0f, 0.0f, 1.0f).Normalize()),
-            Forward(CVector3(1.0f, 0.0f, -1.0f).Normalize()),
-            Target(),
-            LensFocalLength(0.02f),
-            MotionSensitivity(0.005),
-            RotationSensitivity(0.01) {
-            CalculateYFieldOfView();
-         }
-
          /** Initialize from XML */
          void Init(TConfigurationNode& t_tree);
-         /** Rotation around the local Y axis */
-         void RotateUpDown(const CRadians& c_angle);
-         /** Rotation around the local Z axis */
-         void RotateLeftRight(const CRadians& c_angle);
-         /** Rotation around the global Z axis */
-         void RotateLeftRight2(const CRadians& c_angle);
-         /** c_delta is expressed in the camera local coordinates */
-         void Translate(const CVector3& c_delta);
-         /** Places this camera in the right position */
-         void Do();
          /** Calculates the value of YFieldOfView */
          void CalculateYFieldOfView();
-         /** Calculate the sensitivity of the camera */
-         void CalculateSensitivity();
       };
-
-      struct STimelineItem {
-         /** Index of the placement */
-         UInt32 Idx;
-         /** Time step to switch to the indicated placement */
+ 
+      struct SKeyframe {
```

Comment:
Missing documentation. This is an important data structure and it should be documented to help demystify usage

Reply:
Fixed in 1588d020fc01cde3cc3ef7750bbbb8b200b85b44